plugins/catalog/src/js/repositories/components/AddRepositoryFormModal.js

```diff
@@ -79,14 +79,18 @@ class AddRepositoryFormModal extends React.Component {
   }
 
   getButtonDefinition() {
+    const { pendingRequest } = this.props;
+
     return [
       {
         text: i18nMark("Cancel"),
         className: "button button-primary-link flush-left",
         isClose: true
       },
       {
-        text: i18nMark("Add Repository"),
+        text: pendingRequest
+          ? i18nMark("Adding...")
+          : i18nMark("Add Repository"),
         className: "button button-primary",
         isSubmit: true
```

Comment:
Can buttons using this declarative system be disabled?

Reply:
Yes, the confirm modal takes a `disabled` prop, which disables the buttons